### PR TITLE
Fix performance text in encyclopedia and sleep times in production buildings where the workers need some experience

### PR DESCRIPTION
**Pull request by gpalino**
Discussion in https://www.widelands.org/forum/topic/5729/

### CODE REVIEW

File: data/tribes/initialization/barbarians/units.lua [hunk 1]
```diff
@@ -1153,8 +1153,12 @@ wl.Descriptions():new_tribe {
...
             -- TRANSLATORS: Note helptext for a barbarian production site: Lumberjack_s Hut
             note = pgettext("barbarians_building", "The lumberjack’s hut needs trees to fell within the work area."),
-            -- TRANSLATORS: Performance helptext for a barbarian production site: Lumberjack_s Hut
-            performance = pgettext("barbarians_building", "The lumberjack needs %s to fell a tree, not counting the time he needs to reach the destination and go home again."):bformat(ngettext("%d second", "%d seconds", 12):bformat(12))
+            performance = {
+               -- TRANSLATORS: Performance helptext for a barbarian production site: Lumberjack_s Hut
+               pgettext("barbarians_building", "The lumberjack needs %s to fell a tree, not counting the time he needs to reach the destination and go home again."):bformat(ngettext("%d second", "%d seconds", 17):bformat(17)),
+               -- TRANSLATORS: Performance helptext for a barbarian production site: Lumberjack_s Hut
+               pgettext("barbarians_building", "He rests in the hut for %s."):bformat(ngettext("%d second", "%d seconds", 20):bformat(20))
```
Comment (hessenfarmer) [inline data/tribes/initialization/barbarians/units.lua]:
```suggestion
               pgettext("barbarians_building", "Afterwards he rests in the hut for %s."):bformat(ngettext("%d second", "%d seconds", 20):bformat(20))
```

File: data/tribes/initialization/barbarians/units.lua [hunk 2]
```diff
@@ -1449,7 +1453,9 @@ wl.Descriptions():new_tribe {
             -- TRANSLATORS: Lore author helptext for a barbarian production site: Barracks
             lore_author = pgettext("barbarians_building", "Famous barbarian recruitment poster"),
             -- TRANSLATORS: Purpose helptext for a barbarian production site: Barracks
-            purpose = pgettext("barbarians_building", "Equips recruits and trains them as soldiers.")
+            purpose = pgettext("barbarians_building", "Equips recruits and trains them as soldiers."),
+            -- TRANSLATORS: Performance helptext for a barbarian production site: Barracks
+            performance = pgettext("barbarians_building", "The barracks needs %s on average to recruit one soldier."):bformat(ngettext("%d second", "%d seconds", 30):bformat(30))
```
Comment (hessenfarmer) [inline data/tribes/initialization/barbarians/units.lua]:
```suggestion
            performance = pgettext("barbarians_building", "The barracks need %s on average to recruit one soldier."):bformat(ngettext("%d second", "%d seconds", 30):bformat(30))
```
Although a single building this has to be used with plural forms, I believe

**Review by hessenfarmer — DISMISSED** [review]
looks good.
2 small things inline.
would you like to add other tribes/sites here too or in a separate PR

**Comment by gpalino** [comment]
I plan to finalize all barbarians and atlanteans buildings in a reasonable time, not sure if will have capacity for next 3 tribes.

**Comment by hessenfarmer** [comment]
> I plan to finalize all barbarians and atlanteans buildings in a reasonable time, not sure if will have capacity for next 3 tribes.

in this PR or ina a separate one

**Comment by gpalino** [comment]
> 

Barbarians and atlanteans in this PR

**Comment by hessenfarmer** [comment]
> > 
> 
> Barbarians and atlanteans in this PR

ok, ping me when you re ready. next time it might be good to open a PR in draft status if it isn't finished yet, but shall be discussed

**Comment by gpalino** [comment]
One question - I've noticed helper functions in data/tribes/scripting/help/time_strings.lua, used for empire buildings.
Instead of
performance = pgettext("barbarians_building", "The carrier needs %s to get one bucket full of water."):bformat(ngettext("%d second", "%d seconds", 40):bformat(40))
Shorter form is used:
performance = pgettext("empire_building", "The innkeeper needs %1% on average to prepare a ration."):bformat(format_seconds(36))

IMO second form is more readable and maintainable.
Is there anything against usage of second form, for each units.lua?

**Comment by hessenfarmer** [comment]
> One question - I've noticed helper functions in data/tribes/scripting/help/time_strings.lua, used for empire buildings. Instead of performance = pgettext("barbarians_building", "The carrier needs %s to get one bucket full of water."):bformat(ngettext("%d second", "%d seconds", 40):bformat(40)) Shorter form is used: performance = pgettext("empire_building", "The innkeeper needs %1% on average to prepare a ration."):bformat(format_seconds(36))
> 
> IMO second form is more readable and maintainable. Is there anything against usage of second form, for each units.lua?

on the contrary. It is highly wishable to use these helper functions for all helptexts. So if you would like to change the barbarian ones this would be highly appreciated.

**Comment by gpalino** [comment]
> > One question - I've noticed helper functions in data/tribes/scripting/help/time_strings.lua, used for empire buildings. Instead of performance = pgettext("barbarians_building", "The carrier needs %s to get one bucket full of water."):bformat(ngettext("%d second", "%d seconds", 40):bformat(40)) Shorter form is used: performance = pgettext("empire_building", "The innkeeper needs %1% on average to prepare a ration."):bformat(format_seconds(36))
> > IMO second form is more readable and maintainable. Is there anything against usage of second form, for each units.lua?
> 
> on the contrary. It is highly wishable to use these helper functions for all helptexts. So if you would like to change the barbarian ones this would be highly appreciated.

I can change with some script, but probably outside of previous changes. So what about following?
- reviewing & merging this PR in current state
- I'll create new PR just for switching to helper functions
- Afterwards 3rd draft PR, where I can slowly continue with encyclopedia updates

To the 3rd point - when doing work in forked repository, does it matter if it is master or different branch, for the purpose of merging to upstream?

**Comment by hessenfarmer** [comment]
> > > One question - I've noticed helper functions in data/tribes/scripting/help/time_strings.lua, used for empire buildings. Instead of performance = pgettext("barbarians_building", "The carrier needs %s to get one bucket full of water."):bformat(ngettext("%d second", "%d seconds", 40):bformat(40)) Shorter form is used: performance = pgettext("empire_building", "The innkeeper needs %1% on average to prepare a ration."):bformat(format_seconds(36))
> > > IMO second form is more readable and maintainable. Is there anything against usage of second form, for each units.lua?
> > 
> > 
> > on the contrary. It is highly wishable to use these helper functions for all helptexts. So if you would like to change the barbarian ones this would be highly appreciated.
> 
> I can change with some script, but probably outside of previous changes. So what about following?
> 
>     * reviewing & merging this PR in current state
> 
>     * I'll create new PR just for switching to helper functions
> 
>     * Afterwards 3rd draft PR, where I can slowly continue with encyclopedia updates

very well let's do it so. But the most important fix would be for the frisians to apply the same corrections for the 10 secs as for the barbarians.      
 
> 
> To the 3rd point - when doing work in forked repository, does it matter if it is master or different branch, for the purpose of merging to upstream?      
   
It does not need to be master in the forked repository and it is even recommendable to give the branch in the fork a telling name. some developer prefix their branches with the issue number it tends to fix for example, but that is not a requirement.

**Review by hessenfarmer — APPROVED** [review]
LGTM. 
thanks.

**Pull request merged**